receiver/jaegerlegacyreceiver/trace_receiver_test.go

```diff
@@ -0,0 +1,200 @@
+	nowPlus10min2sec := now.Add(d10min).Add(d2sec)
+
+	want := expectedTraceData(now, nowPlus10min, nowPlus10min2sec)
+	batch, err := jaegertranslator.OCProtoToJaegerThrift(want[0])
+	assert.NoError(t, err, "should not have failed proto/thrift translation")
+
+	//confirm port is open before attempting
+	err = testutils.WaitForPort(t, port)
+	assert.NoError(t, err, "WaitForPort failed")
+
+	err = p.GetReporter().EmitBatch(batch)
+	assert.NoError(t, err, "should not have failed to emit batch")
+
+	got := sink.AllTraces()
+	assert.Len(t, batch.Spans, len(want[0].Spans), "got a conflicting amount of spans")
+
+	if diff := cmp.Diff(got, want); diff != "" {
+		t.Errorf("Mismatched responses\n-Got +Want:\n\t%s", diff)
+	}
+}
+
+func expectedTraceData(t1, t2, t3 time.Time) []consumerdata.TraceData {
+	traceID := []byte{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80}
+	parentSpanID := []byte{0x1F, 0x1E, 0x1D, 0x1C, 0x1B, 0x1A, 0x19, 0x18}
+	childSpanID := []byte{0xAF, 0xAE, 0xAD, 0xAC, 0xAB, 0xAA, 0xA9, 0xA8}
+
+	return []consumerdata.TraceData{
+		{
+			Node: &commonpb.Node{
+				ServiceInfo: &commonpb.ServiceInfo{Name: "issaTest"},
+				LibraryInfo: &commonpb.LibraryInfo{},
+				Identifier:  &commonpb.ProcessIdentifier{},
+				Attributes: map[string]string{
+					"bool":   "true",
+					"string": "yes",
+					"int64":  "10000000",
+				},
+			},
+			Spans: []*tracepb.Span{
+				{
+					TraceId:      traceID,
+					SpanId:       childSpanID,
+					ParentSpanId: parentSpanID,
+					Name:         &tracepb.TruncatableString{Value: "DBSearch"},
+					StartTime:    TimeToTimestamp(t1),
+					EndTime:      TimeToTimestamp(t2),
+					Status: &tracepb.Status{
+						Code:    trace.StatusCodeNotFound,
+						Message: "Stale indices",
+					},
+					Attributes: &tracepb.Span_Attributes{
+						AttributeMap: map[string]*tracepb.AttributeValue{
+							"error": {
+								Value: &tracepb.AttributeValue_BoolValue{BoolValue: true},
+							},
+						},
+					},
+					Links: &tracepb.Span_Links{
+						Link: []*tracepb.Span_Link{
+							{
+								TraceId: traceID,
+								SpanId:  parentSpanID,
+								Type:    tracepb.Span_Link_PARENT_LINKED_SPAN,
+							},
+						},
+					},
+				},
+				{
+					TraceId:   traceID,
+					SpanId:    parentSpanID,
+					Name:      &tracepb.TruncatableString{Value: "ProxyFetch"},
+					StartTime: TimeToTimestamp(t2),
+					EndTime:   TimeToTimestamp(t3),
+					Status: &tracepb.Status{
+						Code:    trace.StatusCodeInternal,
+						Message: "Frontend crash",
+					},
+					Attributes: &tracepb.Span_Attributes{
+						AttributeMap: map[string]*tracepb.AttributeValue{
+							"error": {
+								Value: &tracepb.AttributeValue_BoolValue{BoolValue: true},
+							},
+						},
+					},
+				},
+			},
+			SourceFormat: "jaeger",
+		},
+	}
+}
+
+// TimeToTimestamp converts a time.Time to a timestamp.Timestamp pointer.
+func TimeToTimestamp(t time.Time) *timestamp.Timestamp {
```

Comment:
Note, this was copied from https://github.com/open-telemetry/opentelemetry-collector/blob/master/internal/internal.go#L26-L35 as it didn't like trying to import the internal package.

Reply:
yeah, we need to see some of these common functions and move to the translation package.